### PR TITLE
Provide more sense of progress via the loading indicator.

### LINKED ISSUE #2585: improve the spinner
The spinner should provide a better feedback to users, give the impression that something is progressing. A progression bar with some messages could help visualizing that something is happening. Also, the "connecting to model" message does not make sense when landing on a profile or store page, being disconnected from models.

### PR DESCRIPTION
**Pull request by kadams54**
Fixes #2585.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6908/

**Comment by kadams54** [comment]
## QA

Should be tested in both gijoe (`make dist && juju upgrade-gui dist/jujugui-2.4.3.tar.bz2`) and gisf (run against `guiproxy`):

1. Pay particular attention to the loading screen when opening the GUI up; it now cycles through a number of different messages and each message has its own progress bar.
2. Ensure that you don't see the loading screen at any other time: when switching between models, both with the model switcher and via the profile, when going to the store, after going to /new as anonymous, deploying, then logging in, etc.

File: jujugui/templates/index.html.go [hunk 1]
```diff
@@ -179,15 +179,71 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
       var flags = {}; // Declare an empty set of feature flags.
 
+      class MessageRotator {
```
Comment (hatched) [inline jujugui/templates/index.html.go]: Can you add a comment to each file saying that any updates also need to be applied to the other.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6913/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6923/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6927/

**Comment by kadams54** [comment]
Size increased to a percentage and moved to center vertically. @Lukewh and I looked at it both large and on an iPhone 5 and it scaled in both directions quite nicely. Thanks! :shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6928/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**